### PR TITLE
#4721 Add titles, descriptions and icons to the user extensions

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
Connect user extensions to pluginsConfig.json metadata.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#4721

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
I also improved (fixed) `localizedProps` enhancer. Now it also localize arrays

### CODE REVIEW

File: web/client/plugins/userExtensions/ExtensionsPanel.jsx [hunk 1]
```diff
@@ -96,9 +99,20 @@ const ExtensionsPanel = ({
...
+            onSelect: (extension) => updateUserPlugin(extension.name, { active: !extension.active })
+        }
+    ),
+    // get plugins configuration
+    withPluginsDefinition(),
+    // localize extensions labels
+    localizedProps('extensions', 'title'),
+    localizedProps('extensions', 'description'),
+    // add filter state handling
+    withState('filterText', 'onFilter')
```
Comment (MV88) [inline web/client/plugins/userExtensions/ExtensionsPanel.jsx]: Why this change? Wasn't the hook good enough?

File: web/client/plugins/userExtensions/withPluginsDefinition.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+const loadingState = require('../../components/misc/enhancers/loadingState');
+
+import getPluginsConfig from '../../observables/config/getPluginsConfig';
+/**
+ * Adds to user extensions the plugins definition and maps the properties
+ * needed (title, description and glyph) to correctly set labels and symbology,
+ * accordingly with the context manager.
+ */
+export default (pluginsConfigURL) => compose(
+    mapPropsStream(props$ =>
```
Comment (MV88) [inline web/client/plugins/userExtensions/withPluginsDefinition.js]: honestly, the less we use mapPropsStream the better it is, but that's my opinion

File: web/client/plugins/userExtensions/ExtensionsPanel.jsx (hunk 1 above)
Comment (offtherailz) [reply web/client/plugins/userExtensions/ExtensionsPanel.jsx]: It was trowing errors due to wrong usage of hooks (you're violating hooks rules") that I couldn't understand. So I came back to a well known solution.

File: web/client/plugins/userExtensions/withPluginsDefinition.js (hunk 2 above)
Comment (offtherailz) [reply web/client/plugins/userExtensions/withPluginsDefinition.js]: There are many of them in MapStore. we can stop using them after a discussion, but is not a decision to take during a PR review
